avm/utilities/pipelines/platform/Set-AvmGitHubPrLabels.ps1

```diff
@@ -0,0 +1,73 @@
+
+.PARAMETER RepoRoot
+Optional. Path to the root of the repository.
+
+.PARAMETER PrUrl
+Mandatory. The URL of the GitHub pull request, like 'https://github.com/Azure/bicep-registry-modules/pull/2540'
+
+.EXAMPLE
+Set-AvmGitHubPrLabels -Repo 'Azure/bicep-registry-modules' -PrUrl 'https://github.com/Azure/bicep-registry-modules/pull/2540'
+
+.NOTES
+Will be triggered by the workflow avm.platform.set-avm-github-pr-labels.yml
+#>
+function Set-AvmGitHubPrLabels {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $Repo,
+
+        [Parameter(Mandatory = $true)]
+        [string] $PrUrl,
+
+        [Parameter(Mandatory = $false)]
+        [string] $RepoRoot = (Get-Item -Path $PSScriptRoot).parent.parent.parent.parent.FullName
+    )
+
+    # Loading helper functions
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-GithubPrRequestedReviewerTeamNames.ps1')
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-GithubTeamMembersLogin.ps1')
+    . (Join-Path $RepoRoot 'avm' 'utilities' 'pipelines' 'platform' 'helper' 'Get-AvmCsvData.ps1')
+
+    $pr = gh pr view $PrUrl.Replace('api.', '').Replace('repos/', '') --json 'author,title,url,body,comments' --repo $Repo | ConvertFrom-Json -Depth 100
+    $allTeamNames = [array](Get-GithubPrRequestedReviewerTeamNames -PrUrl $PrUrl.Replace('api.', '').Replace('repos/', ''))
+    $teamNames = [array]($allTeamNames | Where-Object { $_ -ne 'bicep-admins' -and $_ -ne 'avm-core-team-technical-bicep' -and $_ -ne 'avm-module-reviewers-bicep' })
+
+    # core team is already assigned, no or more than one module reviewer team is assigned
+    if ($allTeamNames.Contains('avm-core-team-technical-bicep') -or $teamNames.Count -eq 0 -or $teamNames.Count -gt 1) {
+        gh pr edit $pr.url --add-label 'Needs: Core Team :genie:' --repo $Repo
+    } else {
+        $teamMembers = [array](Get-GithubTeamMembersLogin -OrgName $Repo.Split('/')[0] -TeamName $teamNames[0])
+
+        # no members in module team or only one and that user submitted the PR
+        if (($teamMembers.Count -eq 0) -or ($teamMembers.Count -eq 1 -and $teamMembers[0] -eq $pr.author.login)) {
+            gh pr edit $pr.url --add-label 'Needs: Core Team :genie:' --repo $Repo
+        } else {
+            gh pr edit $pr.url --add-label 'Needs: Module Owner :mega:' --repo $Repo
+        }
+
+        # check for orphanded module
+        $moduleName = $teamNames[0]
+        $moduleIndex = $moduleName.StartsWith('avm-res') ? 'Bicep-Resource' : ($moduleName.StartsWith('avm-ptn') ? 'Bicep-Pattern' : 'Bicep-Utility')
+        # get CSV data
+        $modules = Get-AvmCsvData -ModuleIndex $moduleIndex
+
+        foreach ($module in $modules) {
+            if ($module.ModuleName.Replace('-', '').Replace('/', '-') -eq $moduleName) {
+                break
+            }
+        }
+
+        if ($module.ModuleStatus -eq 'Orphaned :eyes:') {
```

Comment:
This seems to be out of the foreach loop of row 63 so `$module.` should not be accessible

Reply:
@eriqua fixed. Although PS somehow still preserved the variable, because it worked.

Reply:
Looks good. I see you removed the break all together. To optimize I'd suggest to keep it but move it after the assignment. Once that is done no need to complete the iteration

Reply:
done